### PR TITLE
feat(r)!: update parser and queries

### PR DESCRIPTION
**Pull request by clason**
Adapt to breaking rewrite @echasnovski

Not sure about the indent changes, and likely all the test will have to be adapted.

### CODE REVIEW

**Comment by echasnovski** [comment]
> Adapt to breaking rewrite @echasnovski
> 
> Not sure about the indent changes, and likely all the test will have to be adapted.

Thanks again for the heads up! I've already adapted in the best possible way: stopped using custom queries :muscle:

**Comment by clason** [comment]
I'm actually pinging you as the listed maintainer for these queries, so I was hoping _you_ would do this work...

**Comment by clason** [comment]
OK, all tests fail, and I don't have the time or the familiarity with R to fix them. Somebody else will have to take this over.

**Comment by echasnovski** [comment]
> I'm actually pinging you as the listed maintainer for these queries, so I was hoping _you_ would do this work...

Sorry, I thought you were being nice :)

------

The fact that I am listed as a maintainer is news to me. It was done [without notifying me](https://github.com/nvim-treesitter/nvim-treesitter/pull/3693#issuecomment-1322396603) (or my consent, really). I indeed [did initial work](https://github.com/nvim-treesitter/nvim-treesitter/commit/773ec8eefc6509dcde1b67f0913e879634dbccf2) but did not list myself as a maintainer on purpose because I did not plan to actively maintain those.

------

@DavisVaughan, are you interested in keeping the R queries up to date here? Or is a recent activity in r-lib/tree-sitter-r has a more localized scope?

**Comment by clason** [comment]
Ah, ok, that's good to know (and sorry about that). I can do two things, though ;)

To be clear, without an active maintainer, I will probably close this PR and demote the parser to "unsupported". We no longer accept unmaintained languages; that is something we can no longer afford (unlike initially, when the project wanted to grow at all costs).

**Comment by ribru17** [comment]
I can volunteer as maintainer

**Comment by lucario387** [comment]
> Not sure about the indent changes, and likely all the test will have to be adapted.

Most lilely the indent queries only. I'll try to see if I can do smth about it.

**Comment by clason** [comment]
No, the highlight tests as well. It looks like the new queries are much "simpler".

**Comment by DavisVaughan** [comment]
The recent merge of `next` into `main` on tree-sitter-r was an accumulation of roughly 2 years of on and off work containing many breaking changes, but also a large amount of improvements and bug fixes. We think `main` should be mostly stable at this point (we will likely have minor improvements and possibly some bug reports from all the changes).

I know nothing about neovim and how it uses tree-sitter, so if @ribru17 or anyone else could be the maintainer then that would be really appreciated. Hopefully it's mostly straightforward to update.

**Comment by echasnovski** [comment]
> I know nothing about neovim and how it uses tree-sitter, so if @ribru17 or anyone else could be the maintainer then that would be really appreciated. Hopefully it's mostly straightforward to update.

It is mostly to keep [queries](https://github.com/nvim-treesitter/nvim-treesitter/tree/master/queries/r) use some standardized capture names and be up to date with latest grammar version. Similar to keeping [built-in queries](https://github.com/r-lib/tree-sitter-r/tree/main/queries) up to date.

**Comment by clason** [comment]
>  Hopefully it's mostly straightforward to update.

Not really; the highlight tests show a number of regressions, and we have indent queries that are (understandably) not tracked upstream.

File: lua/nvim-treesitter/parsers.lua
Comment (admash) [inline lua/nvim-treesitter/parsers.lua:1]: We also need a line with 
branch = "main" 
in install_info = {...}.

Comment (clason) [reply lua/nvim-treesitter/parsers.lua:1]: No, we don't.

Comment (admash) [reply lua/nvim-treesitter/parsers.lua:1]: The default branch is "master" which no longer exists in https://github.com/r-lib/tree-sitter-r ... 
This causes ":TSInstall r" to fail. Am I misunderstanding something?

Comment (clason) [reply lua/nvim-treesitter/parsers.lua:1]: That is a different (and annoying) issue: the working commit we locked to has been removed. So upstream broke the commit we locked to, which means there's currently _no_ working version of this parser. Nothing that can be done until someone steps up and updates the queries and tests.

Comment (clason) [reply lua/nvim-treesitter/parsers.lua:1]: The commit is still there, and `main` _is_ the default branch. `:TSInstall r` works fine for me.

**Comment by ribru17** [comment]
@clason Should I pick this up and fix the highlight tests? If so, should the queries be fixed to pass the tests or should the tests be updated to match the queries?

**Comment by clason** [comment]
> @clason Should I pick this up and fix the highlight tests? If so, should the queries be fixed to pass the tests or should the tests be updated to match the queries?

I (and R users) would be grateful. Somebody who understands R needs to look at the tests and the current parser output to decide whether the changes are OK or inacceptable regressions -- I am guessing the latter, as quite a lot of previous captures no longer exists, but I can't tell.

**Comment by clason** [comment]
Merging this to finally get the update queue unblocked. There are still regressions in some captures and broken indentation after `|>` and `%>%`, but someone else will have to take care of those.

@lucario387 If you have some time, I would appreciate it if you could take a look at the `pipe.R` tests; I don't understand why they fail even though there's a matching query.

**Comment by ribru17** [comment]
Thank you @clason, sorry for the delay. I have been getting busier lately but I can take a look at highlights this weekend :+1:

**Pull request merged**